pybricks/util_pb/pb_flash.c

```diff
@@ -0,0 +1,450 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2021 The Pybricks Authors
+
+
+
+#include "py/mpconfig.h"
+
+#if PYBRICKS_HUB_PRIMEHUB
+
+#include <string.h>
+
+#include <pbio/error.h>
+#include <pbio/util.h>
+
+#include <stm32f4xx_ll_spi.h>
+#include <stm32f4xx_hal.h>
+
+#include "py/mphal.h"
+
+#include "lfs.h"
+
+#include <pybricks/util_pb/pb_flash.h>
+
+SPI_HandleTypeDef hspi2;
+
+// Whether the SPI and filesystem have been initialized once after boot.
+static bool flash_initialized = false;
+
+// True sets the notCS pin, False resets it
+static void flash_enable(bool enable) {
```

Comment:
It there a reason we do this manually instead of setting the pin to AF5 and letting the SPI hardware take care of it automatically?